dotCMS/src/main/java/com/dotcms/rest/api/v1/workflow/WorkflowResource.java

```diff
@@ -0,0 +1,544 @@
+            response = (e.getCause() instanceof SecurityException)?
+                    ExceptionMapperUtil.createResponse(e, Response.Status.UNAUTHORIZED) :
+                    ExceptionMapperUtil.createResponse(e, Response.Status.INTERNAL_SERVER_ERROR);
+        }
+
+        return response;
+    } // findActionByStep.
+
+    /**
+     * Returns a set of actions associated to the schemeId
+     * @param request  HttpServletRequest
+     * @param schemeId String
+     * @return Response
+     */
+    @GET
+    @Path("/schemes/{schemeId}/actions")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public final Response findActionsByScheme(@Context final HttpServletRequest request,
+                                              @PathParam("schemeId") final String schemeId) {
+
+        final InitDataObject initDataObject = this.webResource.init
+                (null, true, request, true, null);
+        Response response;
+        List<WorkflowAction> actions;
+
+        try {
+
+            Logger.debug(this, "Getting the workflow actions: " + schemeId);
+            actions   = this.workflowHelper.findActionsByScheme(schemeId, initDataObject.getUser());
+            response  = Response.ok(new ResponseEntityView(actions)).build(); // 200
+        } catch (Exception e) {
+
+            Logger.error(this.getClass(),
+                    "Exception on findActionsByScheme, schemeId: " + schemeId +
+                            ", exception message: " + e.getMessage(), e);
+            response = (e.getCause() instanceof SecurityException)?
+                    ExceptionMapperUtil.createResponse(e, Response.Status.UNAUTHORIZED) :
+                    ExceptionMapperUtil.createResponse(e, Response.Status.INTERNAL_SERVER_ERROR);
+        }
+
+        return response;
+    } // findActionsByScheme.
+
+    /**
+     * Saves an action, by default the action is associated to the schema, however if the stepId is set will be automatically associated to the step too.
+     * @param request               HttpServletRequest
+     * @param workflowActionForm    WorkflowActionForm
+     * @return Response
+     */
+    @POST
+    @Path("/actions")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public final Response save(@Context final HttpServletRequest request,
+                               final WorkflowActionForm workflowActionForm) {
+
+        final InitDataObject initDataObject = this.webResource.init
+                (null, true, request, true, null);
+        Response response;
+        WorkflowAction newAction;
+
+        try {
+
+            Logger.debug(this, "Saving new workflow action: " + workflowActionForm.getActionName());
+            newAction = this.workflowHelper.save(workflowActionForm, initDataObject.getUser());
+            response  = Response.ok(new ResponseEntityView(newAction)).build(); // 200
+        } catch (Exception e) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)